diff --git a/i18n/index.js b/i18n/index.js
index 1c4baddcca21f1ce366f22de0f6089c4cf52c58b..d3c88890d1f2999fde8bf4eddd6b039c5fa1e4f9 100644
--- a/i18n/index.js
+++ b/i18n/index.js
@@ -1,5 +1,5 @@
 import { I18N } from 'xshell/i18n/index.js';
-import _dict from './dict.json' assert { type: 'json' };
+import _dict from './dict.json';
 export let i18n = new I18N(_dict);
 const { t, language } = i18n;
 export { t, language };
